File: workspace-ui/src/main/java/dev/sayaya/handbook/client/usecase/WorkspaceRepository.java
package dev.sayaya.handbook.client.usecase;

import dev.sayaya.rx.Observable;

/** 워크스페이스 API 포트. */
public interface WorkspaceRepository {
    Observable<String> create(String name, String description);
    Observable<String> update(String id, String name, String description);
    Observable<Void> delete(String id);
}

File: persist-workspace/src/main/kotlin/dev/sayaya/handbook/usecase/WorkspaceRepository.kt
package dev.sayaya.handbook.usecase

import dev.sayaya.handbook.domain.Workspace
import reactor.core.publisher.Mono

interface WorkspaceRepository {
    fun save(workspace: Workspace): Mono<Workspace>
    fun update(workspace: Workspace): Mono<Workspace>
    fun delete(id: java.util.UUID): Mono<Void>
}
